source/neuropod/bindings/java/src/main/native/com_uber_neuropod_NeuropodTensor.cc

```diff
@@ -0,0 +1,77 @@
+#include "com_uber_neuropod_NeuropodTensor.h"
+
+#include "jclass_register.h"
+#include "neuropod/neuropod.hh"
+#include "utils.h"
+
+#include <memory>
+
+#include <jni.h>
+
+using namespace neuropod::jni;
+
+JNIEXPORT void JNICALL Java_com_uber_neuropod_NeuropodTensor_nativeDoDelete(JNIEnv * env, jobject, jlong handle)
+{
+    try
+    {
+        auto tensor = reinterpret_cast<std::shared_ptr<neuropod::NeuropodValue> *>(handle);
+        (*tensor).reset();
+        delete tensor;
```

Comment:
Does it mean it was originally allocated as:
auto tensor = new shared_ptr(...)

Yeah, because tensors are created by tensor allocators in C++ side, and tensor allocators return shared pointer. There is no way to detach a shared pointer from its underlying object, so I move the shared pointer to heap.

Reply:
I think we can keep it named tensor because the Tensor are always wrapped by a tensor, we never get the chance to access it directly.